crates/wasmparser/src/validator/operators.rs

```diff
@@ -938,17 +938,31 @@ where
         Ok(())
     }
 
-    /// Checks the validity of a common atomic binary operator.
-    fn check_atomic_binary_op(&mut self, memarg: MemArg, op_ty: ValType) -> Result<()> {
+    /// Checks the validity of atomic binary operator on memory.
+    fn check_atomic_binary_memory_op(&mut self, memarg: MemArg, op_ty: ValType) -> Result<()> {
         let ty = self.check_shared_memarg(memarg)?;
+        self.check_atomic_binary_op(op_ty, ty)
+    }
+
+    /// Checks the validity of a common atomic binary operator.
+    fn check_atomic_binary_op(&mut self, op_ty: ValType, ty: ValType) -> Result<()> {
+        // TODO: why did the pre-existing code treat `op_ty` and `ty` as
+        // different types?
```

Comment:
Not the best names but for the previous code one type was the index type of memory and another was the type of the operation.